### PR TITLE
digest64

### PR DESCRIPTION
**Pull request by g11in**
#49 
---------------------------------------
PS: *this work has not been funded or granted by anyone. if you think this adds value to the ecosystem you could tip the author at: [0xb44ddd63d0e27bb3cd8046478b20c0465058ff04](https://etherscan.io/address/0xb44ddd63d0e27bb3cd8046478b20c0465058ff04)*

### CODE REVIEW

File: assembly/index.ts [hunk 1]
```diff
@@ -137,22 +158,23 @@ function hashBlocks(wPtr: usize, mPtr: usize): void {
...
-  // Expand message blocks 17-64
-  for (i = 16; i < 64; i++) {
-    store32(wPtr, i,
-      SIG1(load32(wPtr, i - 2)) +
-      load32(wPtr, i - 7) +
-      SIG0(load32(wPtr, i - 15)) +
-      load32(wPtr, i - 16)
-    );
+  // If mPtr is null, wPtr is assumed to be precomputed
+  if(mPtr){
```
Comment (dapplion) [inline assembly/index.ts:162]: Can you mantain the same style of padding and whitespace in code?

**Review by wemeetagain — APPROVED** [review]
Awesome job @g11in 

Solid 5% speedup (on my local machine)

**Comment by g11in** [comment]
> 
> 
> Awesome job @g11in
> 
> Solid 5% speedup (on my local machine)

thanks but just 5%  ? my local benchmarking gives 20% consistently!

**Review by dapplion — APPROVED** [review]
:tada: Looks great! @wemeetagain is the WASM blob under the size limit to be loaded syncronously?

**Pull request merged**